spec/src/main/asciidoc/architecture.adoc

```diff
@@ -142,6 +142,8 @@ export MP_METRICS_TAGS=app=shop,tier=integration,special=deli\=ver\,y
 
 Global tags and tags registered with the metric are included in the output returned from the REST API.
 
+Global tags MUST NOT be automatically added to the `MetricID` objects, they must be appended to all metrics by the implementation during export.
```

Comment:
This line seems more restrictive than what we're discussing is the intent.  I'd suggest just removing this.  Implementors can add tags whenever they want, so long as they show up in the REST output.

We do need to clarify something in the spec though -- are global tags allowed to be application-specific?  If so, I suggest we say in the spec that impls that support running more than one application either MAY or MUST add the "global" tags prior to metric registration to support tracking metrics separately for each app.

Reply:
+1 to dropping "during export" from the sentence.

Aren't global tags global, seems odd to me to have global tags applying only to a particular application?

Reply:
Global tags should be global (all metrics). How an implementation does that is up to it. App specific metrics can be done some other way. 

FWIW, Don and I stumbled on this possibility because global tags are specified in MP Config, which could be overridden by the app. That would have impacted the tags as re-read automatically every time a MetricID was created.. I don't think the same problem exists if implementations manage global tags their own way.

Reply:
If we allow the impl to squeeze in a tag "under the hood" while registering a metric, the application can run into problems because when it retrieves the metric later, it will have a different `MetricID` than what it was registered with.
That wasn't that much of a problem with the old approach where the tag appeared always within the constructor, but now it will become a problem. Therefore I'd say we should forbid implementations to add tags to the `MetricID`